attestation-agent/attester/src/tdx/mod.rs

```diff
@@ -2,25 +2,51 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 //
-
-use crate::utils::pad;
-
+use super::tsm_report::*;
 use super::Attester;
 use anyhow::*;
 use base64::Engine;
-use log::debug;
-use scroll::Pread;
 use serde::{Deserialize, Serialize};
 use std::path::Path;
-use tdx_attest_rs::{self, tdx_report_t};
 
 mod report;
 
+#[cfg(feature = "tdx-dcap-ioctl")]
+use tdx_attest_rs::{self, tdx_report_t};
+
 const TDX_REPORT_DATA_SIZE: usize = 64;
 const CCEL_PATH: &str = "/sys/firmware/acpi/tables/data/CCEL";
 
+fn tdx_getquote_ioctl_is_available() -> bool {
+    if cfg!(feature = "tdx-dcap-ioctl") {
+        Path::new("/dev/tdx-attest").exists() || Path::new("/dev/tdx-guest").exists()
+    } else {
+        false
+    }
+}
+
+#[allow(unused_variables)]
+fn get_quote_ioctl(report_data: &Vec<u8>) -> Result<Vec<u8>> {
+    cfg_if::cfg_if! {
+        if #[cfg(feature = "tdx-dcap-ioctl")] {
+            let tdx_report_data = tdx_attest_rs::tdx_report_data_t {
+                // report_data.resize() ensures copying report_data to
+                // tdx_attest_rs::tdx_report_data_t cannot panic.
+                d: report_data.as_slice().try_into().unwrap(),
```

Comment:
do you consider using 'map_err' instead of 'unwrap'?

Reply:
It's like this on purpose. See the comment and discussion in #452